File: src/main/kotlin/dev/loveeev/astraTowny/events/resident/ResidentCreateEvent.java
package dev.loveeev.astraTowny.events.resident;

import dev.loveeev.astratowny.objects.Resident;
import lombok.Getter;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.bukkit.event.Cancellable;

@Getter
public class ResidentCreateEvent extends Event implements Cancellable {
    private static final HandlerList handlers = new HandlerList();
    private boolean cancelled;
    private final Resident resident;

    public ResidentCreateEvent(Resident resident) {
        this.resident = resident;
        this.cancelled = false;
    }


    @Override
    public boolean isCancelled() {
        return cancelled;
    }

    @Override
    public void setCancelled(boolean cancelled) {
        this.cancelled = cancelled;
    }

    @Override
    public HandlerList getHandlers() {
        return handlers;
    }

    public static HandlerList getHandlerList() {
        return handlers;
    }
}

File: src/main/kotlin/dev/loveeev/astraTowny/events/resident/ResidentCreateEvent.kt
package dev.loveeev.astratowny.events.resident

import dev.loveeev.astratowny.objects.Resident
import org.bukkit.event.Cancellable
import org.bukkit.event.Event
import org.bukkit.event.HandlerList


class ResidentCreateEvent(val resident: Resident) : Event(), Cancellable {
    private var cancelled = false


    override fun isCancelled(): Boolean {
        return cancelled
    }

    override fun setCancelled(cancelled: Boolean) {
        this.cancelled = cancelled
    }

    override fun getHandlers(): HandlerList {
        return handlerList
    }

    companion object {
        val handlerList: HandlerList = HandlerList()
    }
}